Sim/SimDelphesInterface/src/DelphesSimulation.cpp

```diff
@@ -244,84 +265,89 @@ StatusCode DelphesSimulation::execute() {
-  DelphesSimulation::ConvertParticle( m_allPartOutArray   , allParticles      );
-  DelphesSimulation::ConvertParticle( m_stablePartOutArray, genStableParticles);
-  DelphesSimulation::ConvertParticle( m_partonOutArray    , genPartons        );
-
-  // Reconstructed objects from Delphes
-  const char* name;
-  for ( unsigned int l=0; l<m_outCollections.size() ; l++) {
-
-    if ( m_outCollections[l].find("muons")!= std::string::npos ) {
-      name = m_Delphes->GetString("Branch",m_outCollections[l].c_str() ) ;
-      m_muonOutArray = m_Delphes->ImportArray(name);
-      DelphesSimulation::ConvertParticle( m_muonOutArray, recMuons );
-    }
-    if ( m_outCollections[l].find("electrons")!= std::string::npos ) {
-      name = m_Delphes->GetString("Branch",m_outCollections[l].c_str() ) ;
-      m_electronOutArray = m_Delphes->ImportArray(name);
-      DelphesSimulation::ConvertParticle( m_electronOutArray, recElectrons );
-    }
-    if ( m_outCollections[l].find("photons")!= std::string::npos ) {
-      name = m_Delphes->GetString("Branch",m_outCollections[l].c_str() ) ;
-      m_photonOutArray = m_Delphes->ImportArray(name);
-      DelphesSimulation::ConvertParticle( m_photonOutArray, recPhotons );
-    }
-    if ( m_outCollections[l].find("jets")!= std::string::npos ) {
-      name = m_Delphes->GetString("Branch",m_outCollections[l].c_str() ) ;
-      m_jetOutArray = m_Delphes->ImportArray(name);
-      DelphesSimulation::ConvertJet( m_jetOutArray, recJets );
-    }
-    if ( m_outCollections[l].find("MissingET")!= std::string::npos ) {
-      name = m_Delphes->GetString("Branch",m_outCollections[l].c_str() ) ;
-      m_metOutArray = m_Delphes->ImportArray(name);
-      DelphesSimulation::ConvertMET( m_metOutArray, recMETs );
-    }
-    if ( m_outCollections[l].find("ScalarHT")!= std::string::npos ) {
-      name =  m_Delphes->GetString("Branch",m_outCollections[l].c_str() ) ;
-      m_htOutArray = m_Delphes->ImportArray(name);
-      DelphesSimulation::ConvertParticle( m_htOutArray, recHTs );
-    }
-  }
-
-  // Save particle collections to FCCSw data store
-  m_allPartclsHandle.put(allParticles      );
-  m_genPartonsHandle.put(genPartons        );
-  m_genStablesHandle.put(genStableParticles);
-  for ( unsigned int l=0; l<m_outCollections.size() ; l++){
-    if ( m_outCollections[l].find("muons")    != std::string::npos ) m_recMuonHandle.put(recMuons);
-    if ( m_outCollections[l].find("electrons")!= std::string::npos ) m_recElecHandle.put(recElectrons);
-    if ( m_outCollections[l].find("photons")  != std::string::npos ) m_recPhotHandle.put(recPhotons);
-    if ( m_outCollections[l].find("jets")     != std::string::npos ) m_recJetsHandle.put(recJets);
-    if ( m_outCollections[l].find("mets")     != std::string::npos ) m_recMETHandle.put(recMETs);
-    if ( m_outCollections[l].find("hts")      != std::string::npos ) m_recHTSHandle.put(recHTs);
-  }*/
+  auto genParticles     = new fcc::MCParticleCollection();
```

Comment:
Not sure if this actually leaks memory, but I think it would be better to use `fcc::MCParticleCollection *genParticles(nullptr);` (same below)